api/src/main/java/org/pmiops/workbench/db/dao/WorkspaceDao.java

```diff
@@ -4,10 +4,21 @@
 import org.pmiops.workbench.db.model.User;
 import org.pmiops.workbench.db.model.Workspace;
 import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.jpa.repository.Query;
 
+
+/**
+ * Declaration of automatic query methods for Workspaces. The methods declared here are
+ * automatically interpreted by Spring Data (see README).
+ */
 public interface WorkspaceDao extends CrudRepository<Workspace, Long> {
+  @Query("SELECT w FROM Workspace w WHERE w.workspaceNamespace = ?1 AND w.firecloudName = ?2")
+  Workspace get(String workspaceNamespace, String firecloudName);
+
   List<Workspace> findByWorkspaceNamespace(String workspaceNamespace);
-  Workspace findByWorkspaceNamespaceAndFirecloudName(String workspaceNamespace,
-      String firecloudName);
+
   List<Workspace> findByCreatorOrderByNameAsc(User creator);
+
+  @Query("SELECT w FROM Workspace w WHERE w.approved IS NULL AND w.reviewRequested = true")
```

Comment:
I believe this can be rewritten as:

findByApprovedIsNullAndReviewRequestedTrue()

or if you prefer:

findByApprovedAndReviewRequested(boolean, boolean)

Reply:
Also: you're probably going to want to add an index for this to the database... or this will run slowly when we have a lot of workspaces.

And you probably should add OrderByRpTimeRequested so you get things back in some reasonable order.

Reply:
One more thought... if we had a lot of workspaces under review, this might blow up.

So if you can stomach a longish method name, you might want to do:

findTopByApprovedIsNullAndReviewRequestedTrueOrderByRpTimeRequested(Pageable pageable)

or

findTopByApprovedAndReviewRequestedOrderByRpTimeRequested(boolean, boolean, Pageable)

Then make a Pageable limiting how many get returned to some value (either configured or requested by the client.)

Reply:
Note: you can return Slice<Workspace> if you want the method to return pagination information, and Page<Workspace> if you want the method to return both pagination information and a total count.

Reply:
Thanks -- you saved me figuring out how to camelCaseIze that one.

Added an index. [TIL there are a lot of ICD10 codes about index fingers.] I noticed that the existing index in Java, `idx_workspace_fc_name`, does not match the db or changelog XML files (https://precisionmedicineinitiative.atlassian.net/browse/RW-220) -- does that mean only XML is relevant here?

I think our working assumption is that we'll only have a few workspaces in the queue, wo I'll skip pagination for now. I'm inclined to leave a comment w/ rationale, but not file a ticket, since it's a "if/when this gets painful" not "invariably before launch" kind of thing -- but I can file a ticket if you want that reminder. [I think this addresses your comment below, too.]